tests/test_launcher.py

```
@@ -61,45 +65,44 @@ def test_python_interpreter_discovery_avoids_substring_match(
-    @patch("launch.venv.create")
-    @patch("launch.Path.exists")
+    @patch("setup.launch.shutil.rmtree")
+    @patch("setup.launch.venv.create")
+    @patch("setup.launch.Path.exists")
     def test_create_venv_recreate(self, mock_exists, mock_venv_create, mock_rmtree):
         """Test venv recreation when forced."""
         # Mock exists to return True initially, then False after rmtree
         mock_exists.side_effect = [True, False]
         venv_path = ROOT_DIR / "venv"
-        with patch("launch.logger") as mock_logger:
+        with patch("setup.launch.logger") as mock_logger:
```

Ruff (0.15.1)

[error] Local variable `mock_logger` is assigned to but never used

Remove assignment to unused variable `mock_logger`

(F841)